polymorphia-backend/src/main/java/com/agh/polymorphia_backend/controller/CourseController.java

```diff
@@ -1,25 +1,98 @@
 package com.agh.polymorphia_backend.controller;
 
+import com.agh.polymorphia_backend.dto.request.course_import.CourseDetailsRequestDto;
 import com.agh.polymorphia_backend.dto.response.user_context.AvailableCoursesResponseDto;
+import com.agh.polymorphia_backend.exception.SkipUnexpectedExceptionHandler;
+import com.agh.polymorphia_backend.service.course.CourseDetailsService;
+import com.agh.polymorphia_backend.service.course.CourseImportService;
 import com.agh.polymorphia_backend.service.course.CourseService;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import jakarta.validation.Valid;
 import lombok.AllArgsConstructor;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
 import org.springframework.http.ResponseEntity;
 import org.springframework.security.access.prepost.PreAuthorize;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.multipart.MultipartFile;
 
+import java.io.IOException;
 import java.util.List;
 
 @RestController
 @AllArgsConstructor
 @RequestMapping("/courses")
 public class CourseController {
     private final CourseService courseService;
+    private final CourseDetailsService courseDetailsService;
+    private final CourseImportService courseImportService;
 
     @GetMapping()
-    @PreAuthorize("isAuthenticated()")
+    @PreAuthorize("hasAnyAuthority('STUDENT', 'INSTRUCTOR', 'COORDINATOR')")
     public ResponseEntity<List<AvailableCoursesResponseDto>> getAvailableCourses() {
         return ResponseEntity.ok(courseService.getAvailableCourses());
     }
+
+    @PostMapping("/create")
+    @PreAuthorize("hasAnyAuthority('STUDENT', 'INSTRUCTOR', 'COORDINATOR')")
```

Comment:
```suggestion
    @PreAuthorize("hasAnyAuthority('COORDINATOR')")
```
Applies to all new course management related methods.

Reply:
We decided to leave the opportunity to create a course for instructor as well